lab/lab2/js/part3-application.js

```diff
@@ -27,29 +27,52 @@
 /* =====================
   Define a resetMap function to remove markers from the map and clear the array of markers
 ===================== */
-var resetMap = function() {
+var resetMap = function(markers) {
   /* =====================
     Fill out this function definition
   ===================== */
+      _.each(markers, (obj) => {return map.removeLayer(obj)});
+      _.each(markers, (obj) => {return markers.pop()})
 };
 
 /* =====================
   Define a getAndParseData function to grab our dataset through a jQuery.ajax call ($.ajax). It
   will be called as soon as the application starts. Be sure to parse your data once you've pulled
   it down!
 ===================== */
-var getAndParseData = function() {
-  /* =====================
-    Fill out this function definition
-  ===================== */
-};
+
+var getAndParseData = function(downloadData) {
+var downloadData = $.ajax("https://raw.githubusercontent.com/CPLN692-MUSA611/datasets/master/json/philadelphia-crime-snippet.json")
+.done((downloadData) => {
+    parsed = JSON.parse(downloadData);
+    return parsed;
+    }
+  )
+}
+
 
 /* =====================
   Call our plotData function. It should plot all the markers that meet our criteria (whatever that
   criteria happens to be — that's entirely up to you)
 ===================== */
+var filterData = function(y) {
```

Comment:
Does this filter function work for you and load points to the map?

I think it's extremely close, but I was unable to load points to the map.

But it's possible my query wasn't quite right - perhaps district 24 is too restrictive for my search.

Reply:
![image](https://user-images.githubusercontent.com/31627334/53648201-36b1ca80-3c37-11e9-813d-0862d827cf33.png)

I got this point by the inputs. I suppose my codes work. Thanks for comment!

Reply:
And actually the filter doesn't work. I just interrupted due to time constraint.